_source/_posts/2022-06-06-secure-databases-dotnet.md

```diff
@@ -0,0 +1,650 @@
+
+Next open `ScrubService.cs` and replace the code there with this implementation.
+
+```csharp
+using Okta_CosmosDb.Models;
+
+namespace Okta_CosmosDb.Services
+{
+    public class ScrubService : IScrubService
+    {
+        public async Task<ScrubResult> ScrubAsync(Person person)
+        {
+            var task = Task.Run(() => { return new ScrubResult(person, new Random().Next(2) == 0); });
+            return await task;
+        }
+    }
+}
+```
+
+This class serves to mock a true scrub service. The method here is `async` because in a real-world environment that would be the case, but since you are just mocking the service you can wrap the login in `Task.Run` to simulate an `async` method.
+
+### Create your models
+
+Next, you will need a couple of data models to facilitate your views and to pass in your application. In your `Models` folder add a file for `ScrubResult.cs` and add the following code to it.
+
+```csharp
+using Newtonsoft.Json;
+
+namespace Okta_CosmosDb.Models
+{
+    public class ScrubResult
+    {
+        [JsonProperty(PropertyName = "id")]
+        public string Id { get; set; }
+
+        [JsonProperty(PropertyName = "person")]
+        public Person Person { get; set; }
+
+        [JsonProperty(PropertyName = "success")]
+        public bool Success { get; set; }
+
+        public ScrubResult(Person person, bool success)
+        {
+            Person = person;
+            Success = success;
+            Id = Guid.NewGuid().ToString();
+        }
+    }
+}
+```
+
+Now add the file and code for `Person.cs` in your `Models` folder.
+
+```csharp
+using Newtonsoft.Json;
+
+namespace Okta_CosmosDb.Models
+{
+    public class Person
+    {
+        public string Name { get; set; }
+
+        [JsonIgnore]
+        public string SSN { get; set; }
+    }
+}
+```
+
+There are a couple of things going on here that you should understand. First, you are going to store the `ScrubResult` object in your Cosmos database. Each property is explicitly given a name using the `JsonProperty` attribute, however you do not need to do this. You do need a field called `id`.  The `JsonProperty` attribute helps keep your C# code consistently [Pascal case](https://techterms.com/definition/pascalcase) while keeping your Cosmos DB property names camel case.  
+
+The real kicker here is that the API is using the `Newtonsoft.Json` package to serialize the objects into JSON strings. This means you can use any of the features that come with the Newtonsoft package to manipulate your data. This is what you did on the `Person` object where you use the `JsonIgnore` attribute to hide the SSN.
+
+While `JsonIgnore` is the simplest way to remove the SSN from this object, there are many other approaches. You could even create a custom attribute using the `Newtonsoft.Json.Serialization.DefaultContractResolver` and hash, encrypt, or otherwise translate the field into something less sensitive.  
```

Comment:
```suggestion
While `JsonIgnore` is the simplest way to remove the SSN from this object, there are many other approaches. You could even create a custom attribute using the `Newtonsoft.Json.Serialization.DefaultContractResolver`, and hash, encrypt, or otherwise translate the field into something less sensitive.  
```